Src/Newtonsoft.Json/Linq/JConstructor.cs

```diff
@@ -164,9 +164,10 @@ public override void WriteTo(JsonWriter writer, params JsonConverter[] converter
         {
             writer.WriteStartConstructor(_name);
 
-            foreach (JToken token in Children())
+            int count = _values.Count;
```

Comment:
I didn't examine this case, but having measured similar cases before, as a rule, it's not faster if `_values` is an array and the compiler can see that (not the case here, and the `foreach` would be just as fast if that was the case anyway), it's not much faster in an `async` or `yield` iterator (but is still a bit, at a cost of another field to the state machine structure), and it's faster otherwise.